controllers/notes.js

```diff
@@ -1,9 +1,38 @@
-exports.getNoteById = (req, res, next) => {
-        res.status(200).json({
-        success: true,
-         msg: `Note with id: ${req.params.id}`
-    });
+/*These controllers, when the endpoints are ChannelSplitterNode, return JSON, or wahtever */
+
+//this file contains the functions of the app
+
+exports.updateNote = (req, res, next) => {
+  Note.findByIdandUpdate(req.params.ide, req.body, {runValidators:true }, ()=>{
+    if (note) {
+      res.render('views/')
+    } else {
+      console.log('This note doesn\'t exist')
     }
+  });
+  //we are oging to have an arr of elements
+  //check composition bc there could be issues updating it since tis an arr
+  //if you are in template a, redirect to template b
+  res.redirect('user/notes');
+
+}
+
+exports.getNote = (req, res, next) => {
+  Note.findByIde(req.params.id)
+  .then(
+    note => res.render('views/edit.hbs', {note:note})
+  ).catch(
+    //next(new Error("custom error")) //and let the middleware handler, the error
+    err => res.render('views/edit.hbs', {error:err})
```

Comment:
Probably here you would like to:

1. Or pass to next() a custom error response as `new ErrorResponse(`Note not found with id ${req.params.id}`, 404)` (you should create it beforehand.
2. Or render an error view, but not the same view as is the operation has being successful